usl/usl_kevm.h

```diff
@@ -0,0 +1,241 @@
+    std::optional<storage_value_t> (*)(account_address_t, storage_key_t);
+using set_account_storage_callback_t = void (*)(account_address_t,
+                                                storage_key_t, storage_value_t);
+
+using get_account_orig_storage_callback_t =
+    std::optional<storage_value_t> (*)(account_address_t, storage_key_t);
+using set_account_orig_storage_callback_t = void (*)(account_address_t,
+                                                     storage_key_t,
+                                                     storage_value_t);
+
+using add_message_callback_t = void (*)(message_t);
+using get_message_callback_t = std::optional<message_t> (*)(message_id_t);
+
+using get_blockhash_callback_t = hash_t (*)(size_t offset);
+
+// Network
+struct network_t {
+  chain_id_t chain_id;
+  add_account_callback_t add_account;
+  remove_account_callback_t remove_account;
+  get_account_balance_callback_t get_account_balance;
+  set_account_balance_callback_t set_account_balance;
+  get_account_code_callback_t get_account_code;
+  set_account_code_callback_t set_account_code;
+  get_account_nonce_callback_t get_account_nonce;
+  set_account_nonce_callback_t set_account_nonce;
+  get_account_storage_callback_t get_account_storage;
+  set_account_storage_callback_t set_account_storage;
+  get_account_orig_storage_callback_t get_account_orig_storage;
+  set_account_orig_storage_callback_t set_account_orig_storage;
+  add_message_callback_t add_message;
+  get_message_callback_t get_message;
+  get_blockhash_callback_t get_blockhash;
+};
+
+// Log
+using log_topic_t = bytes_t;
+using log_topic_list_t = std::vector<log_topic_t>;
+struct log_t {
+  account_address_t account;
+  log_topic_list_t topics;
+  bytes_t data;
+};
+using log_list_t = std::vector<log_t>;
+
+// Accessed Storage List
+struct accessed_storage_t {
+  account_address_t account_address;
+  storage_key_t storage_key;
+};
+using accessed_storage_list_t = std::vector<accessed_storage_t>;
+
+// Substate
+using account_address_list_t = std::vector<account_address_t>;
+struct substate_t {
+  account_address_list_t self_destrcut;
+  log_list_t log;
+  account_address_list_t touched_accounts;
+  ether_t refund;
+  account_address_list_t accessed_accounts;
+  accessed_storage_list_t accessed_storage;
+};
+
+// Status Code
+enum class status_code_t {
+  EVMC_REJECTED,
+  EVMC_INTERNAL_ERROR,
+  EVMC_SUCCESS,
+  EVMC_REVERT,
+  EVMC_FAILURE,
+  EVMC_INVALID_INSTRUCTION,
+  EVMC_UNDEFINED_INSTRUCTION,
+  EVMC_OUT_OF_GAS,
+  EVMC_BAD_JUMP_DESTINATION,
+  EVMC_STACK_OVERFLOW,
+  EVMC_STACK_UNDERFLOW,
+  EVMC_CALL_DEPTH_EXCEEDED,
+  EVMC_INVALID_MEMORY_ACCESS,
+  EVMC_STATIC_MODE_VIOLATION,
+  EVMC_PRECOMPILE_FAILURE,
+  EVMC_NONCE_EXCEEDED
+};
+
+// Result
+using status_code_option_t = std::optional<status_code_t>;
+struct result_t {
+  bytes_t output;
+  status_code_option_t status_code;
+};
+
+// KEVM mode
+enum class mode_t { NORMAL, VMTESTS };
```

Comment:
Don't bother with this. We do not need to expose this to the external api

Reply:
removed